File: ticket-frontend/src/modules/fuzzy/fuzzy-variable-year.ts
import { FuzzyVariableDistributionPart, FuzzyVariableDistributionPartUtils, FuzzyVariableI } from './fuzzy-variable-distribution';

export class FuzzyVariableYear implements FuzzyVariableI {
    varOld: FuzzyVariableDistributionPart;
    varRecent: FuzzyVariableDistributionPart;
    varNew: FuzzyVariableDistributionPart;
    varVeryNew: FuzzyVariableDistributionPart;

    constructor(obj: {
        varOld: FuzzyVariableDistributionPart,
        varRecent: FuzzyVariableDistributionPart,
        varNew: FuzzyVariableDistributionPart,
        varVeryNew: FuzzyVariableDistributionPart
    }) {
        this.varOld = obj.varOld;
        this.varRecent = obj.varRecent;
        this.varNew = obj.varNew;
        this.varVeryNew = obj.varVeryNew;
    }

    static fromObj(obj: any): FuzzyVariableYear {
        return new FuzzyVariableYear({
            varOld: FuzzyVariableDistributionPartUtils.fuzzyVariableDistributionPartFromObj(obj.varOld),
            varRecent: FuzzyVariableDistributionPartUtils.fuzzyVariableDistributionPartFromObj(obj.varRecent),
            varNew: FuzzyVariableDistributionPartUtils.fuzzyVariableDistributionPartFromObj(obj.varNew),
            varVeryNew: FuzzyVariableDistributionPartUtils.fuzzyVariableDistributionPartFromObj(obj.varVeryNew)
        })
    }

    getFuzzyVariableMap(): Record<string, FuzzyVariableDistributionPart> {
        return {
            [FuzzyVariableYearFields.OLD]: this.varOld,
            [FuzzyVariableYearFields.RECENT]: this.varRecent,
            [FuzzyVariableYearFields.NEW]: this.varNew,
            [FuzzyVariableYearFields.VERY_NEW]: this.varVeryNew,
        }
    }

    getFuzzyVariableColorsMap(): Record<string, string> {
        return {
            [FuzzyVariableYearFields.OLD]: 'red',
            [FuzzyVariableYearFields.RECENT]: 'orange',
            [FuzzyVariableYearFields.NEW]: 'yellow',
            [FuzzyVariableYearFields.VERY_NEW]: 'green',
        }
    }

}

export enum FuzzyVariableYearFields {
    OLD = 'OLD',
    RECENT = 'RECENT',
    NEW = 'NEW',
    VERY_NEW = 'VERY_NEW',
}

File: ticket-backend/src/main/java/aics/domain/fuzzy/models/FuzzyVariableYear.java
package aics.domain.fuzzy.models;

import aics.domain.fuzzy.constants.FuzzyVariableYearFields;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.experimental.Accessors;

import java.io.Serializable;
import java.util.HashMap;

@Data
@Accessors(chain = true)
@AllArgsConstructor
public class FuzzyVariableYear implements Serializable, FuzzyVariableI {
    private FuzzyVariableDistributionPart varOld;
    private FuzzyVariableDistributionPart varRecent;
    private FuzzyVariableDistributionPart varNew;
    private FuzzyVariableDistributionPart varVeryNew;

    @Override
    public HashMap<String, FuzzyVariableDistributionPart> getFuzzyVariableMap() {
        HashMap<String, FuzzyVariableDistributionPart> fuzzyVariableMap = new HashMap<>();
        fuzzyVariableMap.put(FuzzyVariableYearFields.OLD.name(), this.varOld);
        fuzzyVariableMap.put(FuzzyVariableYearFields.RECENT.name(), this.varRecent);
        fuzzyVariableMap.put(FuzzyVariableYearFields.NEW.name(), this.varNew);
        fuzzyVariableMap.put(FuzzyVariableYearFields.VERY_NEW.name(), this.varVeryNew);
        return fuzzyVariableMap;
    }
}
